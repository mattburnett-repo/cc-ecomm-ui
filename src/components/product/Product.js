export default function Product (props) {
    const { img, name, description, price } = props;
    return (
        <div>
            <h2>{name} Details</h2>
            <img aria-label="product-image" src={img} alt={name}/>
            <div role="presentation" aria-label="description">Description: {description}</div>
            <div role="presentation" aria-label="price">Price: {price}</div>
            <input aria-label='quantity' placeholder="Enter product quantity" />
            <button aria-label="add-to-cart">Add To Cart</button>
        </div>
    )
}
