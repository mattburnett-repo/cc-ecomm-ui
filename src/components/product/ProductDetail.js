export default function ProductDetail (props) {

    return (
        <div>
            <h2 aria-label="product-detail-heading">{props.name} Details</h2>
            <img aria-label="product-image" src={props.image_url} alt={props.name}/>
            <div role="presentation" aria-label="description">Description: {props.description}</div>
            <div role="presentation" aria-label="price">Price: {props.price}</div>
            <input aria-label='quantity' placeholder="Enter product quantity" />
            <button aria-label="add-to-cart">Add To Cart</button>
        </div>
    )
}
