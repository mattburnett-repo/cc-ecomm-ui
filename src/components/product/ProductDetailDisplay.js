
import { useHistory } from 'react-router-dom'
import { useDispatch } from 'react-redux'

import ItemQuantity from '../../widgets/ItemQuantity'
import { addItemToCurrentCart } from '../../features/carts/currentCartActions'

import { StyledProductDetailDisplay } from '../styles/ProductDetailDisplay.styled'

// USE THIS COMPONENT TO BREAK OUT AND SHOW INFO ABOUT ONE PRODUCT
//      use ProductDisplay to display products in ProductsListingDisplay

export default function ProductDetailDisplay ( props ) {
    const { item } = props

    const history = useHistory()
    const dispatch = useDispatch()

    return (
        <>
            <header>
            <h4>Product Detail Display</h4>
            </header>
        
            <StyledProductDetailDisplay>
                <div id="productImage" role="presentation" aria-label="product-image">
                    <div role="presentation" aria-label='product' key={item.id}></div>
                    <div role="img" id="product-image-url" aria-label='product-image-url'> 
                        <img src={item.image_url} alt={item.name} id={item.id} />
                    </div>                
                </div>
                <div id="productInfo" role="presentation" aria-label='product-info'>
                    <label htmlFor='product-name'>Product Name:</label>
                    <div id="product-name" role='presentation' aria-label='product-name'>{item.name}</div>
                    <label htmlFor='product-description'>Product Description:</label>
                    <div id="product-description" role='presentation' aria-label='product-description'>{item.description}</div>

                    <div role="presentation" aria-label='product-order-controls'>
                        <label htmlFor='product-price'>Price:</label>
                        <div role='presentation' aria-label='product-price'>{item.price}</div>

                        <div role="presentation" aria-label='product-item-quantity'>
                            <ItemQuantity itemId={item.id}/>
                        </div>
                        <button aria-label="add-to-cart" onClick={() => dispatch(addItemToCurrentCart(item))}>Add To Cart</button>
                        <button aria-label="go-back-button" onClick={() => history.goBack()}>Go Back</button>
                    </div>              
                </div>
            </StyledProductDetailDisplay>
        </>
    )
}