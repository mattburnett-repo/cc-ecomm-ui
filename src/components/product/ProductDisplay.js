
// USE THIS COMPONENT TO DISPLAY INDIVIDUAL PRODUCTS IN PRODUCTS LISTING DISPLAY 
//      use ProductDetailDisplay to break out and show info about one product

import { useHistory } from 'react-router-dom'

import OrderControls from '../../widgets/OrderControls'

import { StyledProductDisplay } from "../styles/ProductDisplay.styled"

export default function ProductDisplay (props) {
    const { item } = props

    const history = useHistory()

    return (
        <StyledProductDisplay>
                <div role="img" id="product-image-url" aria-label='product-image-url'> 
                    <img src={item.image_url} alt={item.name} id={item.id} onClick={() => history.push('/product-detail/' + item.id)}/>
                </div>

                <h3 role='presentation' aria-label='product-name'>{item.name}</h3>

                <OrderControls item={item} />
        </StyledProductDisplay>
    )
}
