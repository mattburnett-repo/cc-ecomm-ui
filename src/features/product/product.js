import ProductDetail from '../../components/product/Product'

export default function Product (props) {
    // do stuff like get data

    return <ProductDetail />
}