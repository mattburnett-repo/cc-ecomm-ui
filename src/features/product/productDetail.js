import ProductDetailDisplay from '../../components/product/ProductDetailDisplay'

export default function ProductDetail (props) {
    // data should come from API / Redux
    
    const mockData = {
                        "id": 2,
                        "category_id": 2,
                        "name": "product_name_02",
                        "description": "product_desc_02",
                        "image_url": "http://www.image.com/item/2",
                        "price": "2.00"
                    };

    return <ProductDetailDisplay data={mockData} />
}