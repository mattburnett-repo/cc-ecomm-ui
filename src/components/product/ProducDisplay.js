
// USE THIS COMPONENT TO DISPLAY PRODUCTS IN PRODUCTS LISTING DISPLAY 
//      use ProductDetailDisplay to break out and show info about one product

import { useDispatch } from 'react-redux'
import { useHistory } from 'react-router-dom'

import { addItemToCurrentCart } from "../../features/carts/currentCartActions"
import ItemQuantity from '../../widgets/ItemQuantity'

import { StyledProductDisplay } from "../styles/ProductDisplay.styled"

export default function ProductDisplay (props) {
    const { item } = props

    const dispatch = useDispatch()
    const history = useHistory()

    return (
        <StyledProductDisplay>
            {/* <div role="presentation" aria-label='product' key={item.id}> */}
                <div role="img" id="product-image-url" aria-label='product-image-url'> 
                    <img src={item.image_url} alt={item.name} id={item.id} onClick={() => history.push('/product-detail/' + item.id)}/>
                    {/* <img src={item.image_url} alt={item.name} id={item.id} /> */}
                </div>

                <h3 role='presentation' aria-label='product-name'>{item.name}</h3>

                <div id="product-order-controls" role="presentation" aria-label='product-order-controls'>
                    <label htmlFor='product-price'>Price:</label>
                    <div role='presentation' aria-label='product-price'>{item.price}</div>

                    <div role="presentation" aria-label='product-item-quantity'>
                        <ItemQuantity itemId={item.id}/>
                    </div> &nbsp;&nbsp;
                    <button aria-label="add-to-cart" onClick={() => dispatch(addItemToCurrentCart(item))}>Add To Cart</button>
                </div>
            {/* </div> */}
        </StyledProductDisplay>
    )
}
