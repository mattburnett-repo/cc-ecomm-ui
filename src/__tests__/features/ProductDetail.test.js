import { render, screen } from '@testing-library/react';
import product from '../../features/product/product';

describe('ProductDetail', () => {
    test('it has a product detail feature', () => {

    })

    // it gets a mock data

    // it has a product detail component
})