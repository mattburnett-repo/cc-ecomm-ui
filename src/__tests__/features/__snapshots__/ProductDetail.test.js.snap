// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductDetail /> feature tests renders a snapshot 1`] = `
<div>
  <div
    aria-label="name"
    role="presentation"
  >
    product_name_02
  </div>
  <img
    alt="product_name_02"
    aria-label="product-image"
    src="http://www.image.com/item/2"
  />
  <div
    aria-label="description"
    role="presentation"
  >
    Description: 
    product_desc_02
  </div>
  <div
    aria-label="price"
    role="presentation"
  >
    Price: 
    2.00
  </div>
  <input
    aria-label="quantity"
    placeholder="Enter product quantity"
  />
  <button
    aria-label="add-to-cart"
  >
    Add To Cart
  </button>
</div>
`;
